src/metabase/notification/api/notification.clj

```diff
@@ -164,6 +165,19 @@
     (events/publish-event! :event/notification-create {:object notification :user-id api/*current-user-id*})
     notification))
 
+(api.macros/defendpoint :post "/payload"
+  "Return the payload of a notification"
+  [_route _query body :- ::models.notification/NotificationWithPayload]
+  (api/create-check :model/Notification body)
+  {:payload (notification/notification-payload (cond-> body
+                                                 (= :notification/system-event (:payload_type body))
+                                                 (assoc :event_info (events/event-info-example (-> body :payload :event_name) (:payload body)))))
```

Comment:
nit: You pull out the event name, payload, and payload type twice, I think it's worth a `let` to just bind them at the top